sui_types/src/messages.rs

```diff
@@ -157,7 +130,10 @@ impl SingleTransactionKind {
         // the same shared object doesn't show up more than once in the same single
         // transaction.
         let mut used = HashSet::new();
-        if !input_objects.iter().all(|o| used.insert(o.object_id())) {
+        if input_objects.iter().any(|o| {
+            let was_newly_inserted = used.insert(o.object_id());
+            !was_newly_inserted
+        }) {
```

Comment:
Why this change?

Reply:
I can revert it if you want (I was probably touching this more aggressively in some earlier version of the PR)

Reply:
Reverted